libdnf/rpm/solv/goal_private.hpp

```diff
@@ -173,7 +177,9 @@ inline GoalPrivate & GoalPrivate::operator=(const GoalPrivate & src) {
             transaction_free(libsolv_transaction);
             libsolv_transaction = nullptr;
         }
-        protected_packages.reset(new libdnf::solv::SolvMap(*src.protected_packages));
+        if (src.protected_packages) {
+            protected_packages.reset(new libdnf::solv::SolvMap(*src.protected_packages));
+        }
```

Comment:
And what about the situation when `protected_packages` is `nullptr` in the `src` but not in the `this`.
May be code is missing:
```
        } else {
            protected_packages.reset();
        }
```